commit e0bbf92682ad1df36ef43104a036469ac0ab3a4a
Author: Justin Stitt <justinstitt@google.com>
Date:   Mon Sep 11 17:52:44 2023 +0000

    um,ethertap: Replace deprecated strncpy() with strscpy()
    
    `strncpy` is deprecated for use on NUL-terminated destination strings [1].
    
    `gate_buf` should always be NUL-terminated and does not require
    NUL-padding. It is used as a string arg inside an argv array given to
    `run_helper()`. Due to this, let's use `strscpy` as it guarantees
    NUL-terminated on the destination buffer preventing potential buffer
    overreads [2].
    
    This exact invocation was changed from `strcpy` to `strncpy` in commit
    7879b1d94badb ("um,ethertap: use strncpy") back in 2015. Let's continue
    hardening our `str*cpy` apis and use the newer and safer `strscpy`!
    
    Link: www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings[1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Cc: Kees Cook <keescook@chromium.org>
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20230911-strncpy-arch-um-os-linux-drivers-ethertap_user-c-v1-1-d9e53f52ab32@google.com
    Signed-off-by: Kees Cook <keescook@chromium.org>

diff --git a/arch/um/os-Linux/drivers/ethertap_user.c b/arch/um/os-Linux/drivers/ethertap_user.c
index 9483021d86dd..3363851a4ae8 100644
--- a/arch/um/os-Linux/drivers/ethertap_user.c
+++ b/arch/um/os-Linux/drivers/ethertap_user.c
@@ -105,7 +105,7 @@ static int etap_tramp(char *dev, char *gate, int control_me,
 	sprintf(data_fd_buf, "%d", data_remote);
 	sprintf(version_buf, "%d", UML_NET_VERSION);
 	if (gate != NULL) {
-		strncpy(gate_buf, gate, 15);
+		strscpy(gate_buf, gate, sizeof(gate_buf));
 		args = setup_args;
 	}
 	else args = nosetup_args;